gnucash/gnome-utils/gnc-plugin.c

```diff
@@ -231,155 +223,266 @@ gnc_plugin_get_name (GncPlugin *plugin)
+                {
+                    gtk_tool_button_set_label (GTK_TOOL_BUTTON(tool_item), _(update->action_label));
+                    gtk_tool_button_set_use_underline (GTK_TOOL_BUTTON(tool_item), TRUE);
+                }
+
+                if (update->action_tooltip)
+                {
+                    gtk_tool_item_set_tooltip_text (GTK_TOOL_ITEM(tool_item), _(update->action_tooltip));
+                    gnc_tool_item_setup_tooltip_to_statusbar_callback (GTK_WIDGET(tool_item), statusbar);
+                }
+                PINFO("Tool item %p, label '%s', tooltip '%s', use_short %d",
+                       menu_item, update->action_label, update->action_tooltip, use_short_names);
+
+                gtk_widget_set_visible (GTK_WIDGET(tool_item), TRUE);
+            }
         }
         else
         {
-            g_warning("No such action with name '%s' in action group %s (size %d)",
-                      action_names[i], gtk_action_group_get_name(action_group),
-                      g_list_length(gtk_action_group_list_actions(action_group)));
+            if (menu_item)
+                gtk_widget_set_visible (GTK_WIDGET(menu_item), FALSE);
+            if (tool_item)
+                gtk_widget_set_visible (GTK_WIDGET(tool_item), FALSE);
         }
     }
 }
 
 
-/** Load a new set of actions into an existing UI.
- *
- *  See gnc-plugin.h for documentation on the function arguments. */
-gint
-gnc_plugin_add_actions (GtkUIManager *ui_merge,
-                        GtkActionGroup *action_group,
-                        const gchar *filename)
+void
+gnc_plugin_update_display_menu_items (GHashTable *display_item_hash,
+                                      GtkWidget *menubar,
+                                      GtkWidget *statusbar)
 {
-    GError *error = NULL;
-    gchar *pathname;
-    gint merge_id;
+    GList *display_list;
 
-    g_return_val_if_fail (ui_merge, 0);
-    g_return_val_if_fail (action_group, 0);
-    g_return_val_if_fail (filename, 0);
+    g_return_if_fail (display_item_hash != NULL);
+    g_return_if_fail (menubar != NULL);
 
-    ENTER("ui_merge %p, action_group %p, filename %s",
-          ui_merge, action_group, filename);
-    gtk_ui_manager_insert_action_group (ui_merge, action_group, 0);
+    ENTER("display_item_hash %p, menubar %p", display_item_hash, menubar);
 
-    pathname = gnc_filepath_locate_ui_file (filename);
-    if (pathname == NULL)
+    display_list = gnc_menu_get_items (menubar);
+
+    for (GList *ptr = display_list; ptr; ptr = g_list_next (ptr))
     {
-        LEAVE("fail");
-        return 0;
-    }
+        GncDisplayItem *gdi;
+        GtkWidget *item = GTK_WIDGET(ptr->data);
+        const gchar *action_name = (const gchar*)g_object_get_data (G_OBJECT(item), "myaction-name");
 
-    merge_id = gtk_ui_manager_add_ui_from_file (ui_merge, pathname, &error);
-    DEBUG("merge_id is %d", merge_id);
+        PINFO("action_name is '%s'", action_name);
```

Comment:
I'd put this after the test for a valid action_name below. I'm not sure the log formatter gracefully handles null pointers.

Reply:
OK.